akka-stream/src/main/scala/akka/stream/scaladsl/Flow.scala

```diff
@@ -440,6 +440,36 @@ trait FlowOps[+Out, +Mat] {
     andThen(MapAsyncUnordered(parallelism, f.asInstanceOf[Any ⇒ Future[Any]]))
 
   /**
+   * Transform stream by applying the given function to each of the elements
+   * as they pass through this processing step. The function wraps in `Future` and the
+   * value of that future will be emitted downstreams. As many futures as requested elements by
+   * downstream may run in parallel and each processed element will be emitted dowstream
+   * as soon as it is ready, i.e. it is possible that the elements are not emitted downstream
+   * in the same order as received from upstream.
+   *
+   * If the group by function `f` throws an exception and the supervision decision is
+   * [[akka.stream.Supervision.Stop]] the stream will be completed with failure.
+   *
+   * If the group by function `f` throws an exception the supervision decision is
+   * [[akka.stream.Supervision.Resume]] or [[akka.stream.Supervision.Restart]]
+   * the element is dropped and the stream continues.
+   *
+   * '''Emits when''' any of the functions complete
+   *
+   * '''Backpressures when''' the number of functions wrapped in futures reaches the configured parallelism and the downstream backpressures
+   *
+   * '''Completes when''' upstream completes and all functions has returned values and all elements has been emitted
+   *
+   * '''Cancels when''' downstream cancels
+   *
+   * @see [[#mapAsyncUnordered]]
+   */
+  def foreachParallel[T](parallelism: Int)(f: Out ⇒ T)(implicit ec: ExecutionContext): Repr[T, Mat] =
+    andThen(MapAsyncUnordered(
+      parallelism,
+      { out: Out ⇒ Future(f(out))(ec) }.asInstanceOf[Any ⇒ Future[Any]]))
+
```

Comment:
Issue with foreach is that it fails Future as soon as any of it's elements fails. So perhaps need to create another foreach in Sink that fail only when stream is completed or not fail at all.